### DISCUSSION TITLE
Adding a station specific browser icon

### DISCUSSION THREAD

**Discussion by xaptronic** [AzuraCast/AzuraCast#6857] [Ideas]
Hello, I'm setup and running an installation currently with one station. Installed the public station page as an app on my phone, currently uses the default browser icon. I would like to change this to my station logo so it shows that icon on my phone. It appears I can only do this by overriding the admin browser icon which I gather would set it for all stations. However, if I added a second station I might want to have a unique browser icon from that.

I am technical and took a look at the source code and found these two sets of files:

**Station**
- https://github.com/AzuraCast/AzuraCast/blob/3b8c7e9e00d5d8ab714955a0de8e8681ebb0a984/frontend/src/components/Stations/Branding.vue
- https://github.com/AzuraCast/AzuraCast/blob/main/src/Controller/Api/Stations/CustomAssets/PostCustomAssetAction.php

Admin
- https://github.com/AzuraCast/AzuraCast/blob/3b8c7e9e00d5d8ab714955a0de8e8681ebb0a984/frontend/src/components/Admin/Branding.vue#L25
- https://github.com/AzuraCast/AzuraCast/blob/main/src/Controller/Api/Admin/CustomAssets/PostCustomAssetAction.php#L18

It appears like the way the assets are handled in both cases are the same, and look generic per asset type - would it be that simply adding the a form field for the browser icon to the station branding widget would make this work? - I can answer this via experimentation, however - in terms of if I were to contribute a PR for this, is there any preference for how I go about doing that beyond the basic mechanism of fork / patch / PR?

Thanks - this is a great piece of software!

**Comment by BusterNeece** [comment]
@xaptronic Thank you for your post!

I wouldn't mind a PR to add this functionality at all. In initially building the per-station branding, I figured that per-station browser icons would be more trouble than they're worth (especially given the fact that we have to make a bunch of resized copies of the browser icon at various standardized sizes), but I can see the utility especially if you're pinning the station's player page to your phone's home screen.

If you have time to, feel free to submit a pull request adding the functionality; we can always tweak anything in the PR so don't worry about it being perfect. If you don't, I can probably put it into the bigger schedule, but you'd want to submit it over at https://features.azuracast.com/ just to make sure we don't lose track of it.

**Reply by xaptronic** [reply]
Great, I will make an attempt on this.

Out of curiosity, when I do a `./docker.sh install-dev` the process of (I think it's composer) fetching dependencies can take almost an hour. Is that typical?

I see quite a few messages like

```
Failed to download phpunit/php-file-iterator from dist: curl error 28 while downloading https://api.github.com/repos/sebastianbergmann/php-file-iterator/zipball/a95037b6d9e608ba092da1b23931e537cadc3c3c: Resolving timed out after 10000 milliseconds
```

I wouldn't characterize my internet as slow.. I'm doing about 215/20 mbps down/up.

If you might have any suggestions that would be great - it's been a little while since I worked in the PHP ecosystem.

**Reply by BusterNeece** [reply]
@xaptronic Definitely not sure what's up there. I'd make sure if you're using WSL for dev that you use WSL2 specifically (ensure your Linux distro is set to run with WSL2) and keep AzuraCast entirely within the Linux filesystem, otherwise you'll incur some pretty significant performance hits. It should never take an _hour_ for Composer deps to install...maybe a minute or so.

**Reply by xaptronic** [reply]
Hmm.. I'm running a Manjaro distro.. not using windows (presumably that's Windows Subsystem for Linux 2 that you refer to by WSL2?)

Odd.. I'll some more digging.

**Reply by Moonbase59** [reply]
@xaptronic That _is_ odd. I’ve also tried Arch and Manjaro some time ago, and it should work. Curl error 28 is a connection timeout, so hopefully it was only a temporary connection or DNS resolving problem? (It says "_Resolving_ timed out" in your error message.)

I tried to download `sebastianbergmann-php-file-iterator-4.1.0-0-ga95037b.zip` from here just now, and it took only fractions of a second (on a 100 MBit/s d/l).

**Reply by xaptronic** [reply]
Turns out I had docker and podman installed. I removed podman and now the setup runs as expected (it seems like it was using both tools in the setup, I'm not going to dig further into that for now though).
